client/ayon_deadline/plugins/publish/global/collect_deadline_job_extra_info.py

```diff
@@ -0,0 +1,44 @@
+import json
+import pyblish.api
+
+from ayon_deadline.lib import FARM_FAMILIES, JOB_EXTRA_INFO_DATA_KEY
+from ayon_applications.utils import get_tools_for_context
+
+
+class CollectDeadlineJobExtraInfo(pyblish.api.InstancePlugin):
+    """Collect set of environment variables to submit with deadline jobs"""
+    order = pyblish.api.CollectorOrder + 0.499
+    label = "Deadline Farm Extra Info"
+    families = FARM_FAMILIES
+    targets = ["local"]
+
+    def process(self, instance):
+
+        # Transfer some environment variables from current context
+        job_info = instance.data.setdefault(JOB_EXTRA_INFO_DATA_KEY, {})
```

Comment:
Calling it directly `job_info` seems weird, I would prefer `job_extra_info` which is more descriptive.